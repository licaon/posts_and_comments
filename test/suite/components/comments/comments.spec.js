import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Comments from 'src/components/comments/comments';

describe('Comments component', () => {
  const createComponent = (props) => {
    const wrapper = shallow(
      <Comments
        {...props}
      />);
    return toJson(wrapper);
  };
  it('should render component correctly without comments', (done) => {
    const component = createComponent({
      comments: undefined,
      postId: 1,
    });

    expect(component).toMatchSnapshot();
    done();
  });
  it('should render component correctly with comments', (done) => {
    const component = createComponent({
      comments: [{
        id: 1,
        name: 'name',
        body: 'body',
      }],
      postId: 1,
    });

    expect(component).toMatchSnapshot();
    done();
  });
});
