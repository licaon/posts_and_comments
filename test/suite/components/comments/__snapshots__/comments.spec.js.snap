// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments component should render component correctly with comments 1`] = `
<div
  className="comments"
>
  <Comment
    comment={
      Object {
        "body": "body",
        "id": 1,
        "name": "name",
      }
    }
  />
</div>
`;

exports[`Comments component should render component correctly without comments 1`] = `
<div
  className="comments"
>
  <Loading
    id={1}
  />
</div>
`;
